DNN Platform/DotNetNuke.ModulePipeline/ModuleControlPipeline.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.IO;
+using DotNetNuke.Entities.Modules;
+using DotNetNuke.Instrumentation;
+using DotNetNuke.Services.Localization;
+using DotNetNuke.UI.Modules.Html5;
+using DotNetNuke.UI.Modules;
+using DotNetNuke.Web.Razor;
+using System.Collections.Generic;
+using DotNetNuke.Web.Mvc;
+
+#if NET472
+using System.Web.UI;
+#endif
+
+namespace DotNetNuke.ModulePipeline
+{
+    /// <summary>
+    /// The Module Pipeline that determines which Module pattern
+    /// to invoke based on the input module type.
+    /// </summary>
+    public class ModuleControlPipeline
+        // MULTI-TARGETTING PIPELINE
+        // -------------------------
+        // This file multi-targets .NET Framework and .NET Standard,
+        // which is needed as DNN migrates to .NET Core. The 'NET472'
+        // pre-processor directives are to fully support Legacy DNN.
+        // As the Pipeline is upgraded to be more complaint with 
+        // .NET Standard 2.0 use the apprioprate pre-processor directives.
+#if NET472
+        : IModuleControlPipeline
+#endif
+    {
+        private static readonly ILog TracelLogger = LoggerSource.Instance.GetLogger("DNN.Trace");
+        private Dictionary<string, IModuleControlFactory> _controlFactories;
+        public ModuleControlPipeline(
+            WebFormsModuleControlFactory webforms,
+            Html5ModuleControlFactory html5,
+            RazorModuleControlFactory razor3,
+            MvcModuleControlFactory mvc,
+            ReflectedModuleControlFactory fallthrough)
+        {
+            _controlFactories = new Dictionary<string, IModuleControlFactory>();
+            _controlFactories.Add(".ascx", webforms);
+            _controlFactories.Add(".htm", html5);
+            _controlFactories.Add(".html", html5);
+            _controlFactories.Add(".cshtml", razor3);
+            _controlFactories.Add(".vbhtml", razor3);
+            _controlFactories.Add(".mvc", mvc);
+            _controlFactories.Add("default", fallthrough);
+        }
+
+#if NET472
+        private IModuleControlFactory GetModuleControlFactory(string controlSrc)
+        {
+            string extension = Path.GetExtension(controlSrc.ToLowerInvariant());
```

Comment:
Consider giving the dictionary a `StringComparer.OrdinalIgnoreCase` argument (on line 43), so that we don't need to worry about casing here

Reply:
Good idea, I'll update this